src/mscorlib/src/System/Reflection/AssemblyName.cs

```diff
@@ -533,7 +571,83 @@ internal string GetNameWithPublicKey()
         private extern byte[] nGetPublicKeyToken();
     
         [System.Security.SecurityCritical]  // auto-generated
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        static internal extern String EscapeCodeBase(String codeBase);
+        static internal String EscapeCodeBase(String codebase)
+        {
+            return EscapeString(codebase, false, true, true);
```

Comment:
I tried on desktop clr #, [, & ] seems to be converted to corresponding escape sequences and matches what mono does. You are correct about %hi though.